tools/build_api.py

```diff
@@ -312,9 +313,14 @@ def prepare_toolchain(src_paths, target, toolchain_name,
 
     # Toolchain instance
     try:
-        toolchain = TOOLCHAIN_CLASSES[toolchain_name](
-            target, options, notify, macros, silent,
-            extra_verbose=extra_verbose)
+        if toolchain_name == "GCC_ARM":
```

Comment:
This is a bit weird. It makes `build_api.py` aware of a detail (the toolchain name) that is internal to the toolchain implementation. If that toolchain name changes for any reason in the toolchain implementation, this code will break silently. Not a huge problem in practice probably, since it's unlikely that the tool name would ever change, but still a bit weird. I would've added the `coverage_filter` argument to the constructor of the base toolchain class (`mbedToolchain`) and let each toolchain implementation worry about it separately.